librz/core/cvfile.c

```diff
@@ -0,0 +1,168 @@
+// SPDX-FileCopyrightText: 2021 Florian Märkl <info€florianmaerkl.de>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_core.h>
+
+#define URI_SCHEME     "vfile://"
+#define URI_SCHEME_LEN 8
+
+typedef struct {
+	RzBinFile *bf;
+	RzBinVirtualFile *vfile;
+	ut64 off;
+} VFileCtx;
+
+extern RzIOPlugin rz_core_io_plugin_vfile;
+
+static bool __check(RzIO *io, const char *pathname, bool many) {
+	return !strncmp(pathname, URI_SCHEME, URI_SCHEME_LEN);
+}
+
+static RzIODesc *__open(RzIO *io, const char *pathname, int rw, int mode) {
+	if (strncmp(pathname, URI_SCHEME, URI_SCHEME_LEN)) {
```

Comment:
reuse `__check`?

Reply:
The thing here is that right after this, I do `pathname + URI_SCHEME_LEN` so it relies on the fact that pathname is at least `URI_SCHEME_LEN` long. If I would use the check function here and the check function is updated later, this consequence may be forgotten.

Reply:
mmm isn't this exactly what `__check` shall do? I mean, IIRC check is called before calling open, so keeping the condition only in check seems to be the best thing to do to avoid issues in case things are updated later.

Reply:
I think it's more of an implementation detail that `check` is always called before opening. I could imagine forcing another io plugin using C api but I think that's also a question of the IO API design, so it's debatable.

Reply:
I didn't mean to rely on the fact that `vf_check` is already called, but that this check and the check in `vf_check` should be essentially the same I think.

Reply:
I've changed it to use `vf_check` now and added a comment in the `vf_check` body.